### PR TITLE
feat: add Luadoc

### PR DESCRIPTION
**Pull request by amaanq**
Mix of EmmyLua and what I noticed commonly seen in lua repos (and what lua_ls recognized)

### CODE REVIEW

File: queries/luadoc/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  "@see"
+  "extends"
+  (diagnostic_identifier)
+] @keyword
+
+[
+  "@async"
+] @keyword.coroutine
+
+(language_injection "@language" (identifier) @keyword)
```
Comment (ObserverOfTime) [inline queries/luadoc/highlights.scm:26]: Shouldn't the identifier be something else?

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

File: queries/luadoc/highlights.scm (hunk 1 above)
Comment (amaanq) [reply queries/luadoc/highlights.scm:26]: I couldn't think of a better fit, but yeah I agree. Here's an example

```lua
---@language JSON
local jsonText = [[{
    "name":"Emmy"
}]]
```

Comment (ObserverOfTime) [reply queries/luadoc/highlights.scm:26]: Maybe `@constant`?

Comment (amaanq) [reply queries/luadoc/highlights.scm:26]: Imo that doesn't feel exactly right, but is ok (another one could be `@type` but that also feels not exactly right), I'll just go w/ that

Comment (ObserverOfTime) [reply queries/luadoc/highlights.scm]: Could just use `@parameter`.

Comment (amaanq) [reply queries/luadoc/highlights.scm]: Yeah, that would fit better and be more consistent in highlighting with other annotations, that didn't cross my mind - thanks

Comment (amaanq) [reply queries/luadoc/highlights.scm]: Forgot to push - fixed now

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**